pycbc/workflow/core.py

```diff
@@ -198,7 +198,10 @@ def __init__(self, cp, name,
             raise TypeError("Failed to find %s executable " 
                             "at %s" % (name, exe_path))
         
-        self.add_pfn(exe_path)
+        if exe_path.startswith('gsiftp://'):
+            self.add_pfn(exe_path,site='pycbc-code')
```

Comment:
I agree that ultimately the weirdness is in the Pegasus symlink handling. Symlinking was never a first-class feature in the design of Pegasus; it was added (years ago) to handle the LIGO case where we pre-stage frame files to a particular site. I'm not claiming that the way that Pegasus currently handles it is correct. There are a couple of other bigger issues in Pegasus' data model that I think need revision (for example whether you have a shared file system or not, or use condor i/o is defined at the workflow level, not the site level, as it should be. The pool attribute in the static PFN cache was added as a work around for the frame data, but we now use it for a lot more.

We should have a face to face with the Pegasus team to hash out some of these issues longer term. For now, I'll just change this to nonlocal for symmetry with other use cases.